qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -87,10 +87,16 @@ def _choose_layout_condition(property_set):
     if backend_properties:
         _choose_layout = NoiseAdaptiveLayout(backend_properties)
 
-    # 3. Extend dag/layout with ancillas using the full coupling map
+    # 3. Remove reset gates if qubit is in zero state and diagonal gates before measurement
+    def _remove_gates_condition(property_set):
```

Comment:
`_remove_gates_condition` isn't needed here because we're adding these passes outside the "run until fixed depth" optimization loop.